src/scripts/main.js

```diff
@@ -1,11 +1,25 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve, reject) => {
+    element.addEventListener(
+      eventName,
+      (e) => {
+        resolve(
+          `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+        );
+      },
+      { once: true },
+    );
+  });
 }
 
 const printMessage = (message) => {
-  // write your code here
+  const div = document.createElement('div');
+
+  div.classList.add('message');
+  div.innerText = message;
```

Comment:
Directly manipulating the `innerText` property can lead to security issues such as XSS if the message contains untrusted content. Ensure that `message` is properly sanitized, or use `textContent` instead if the content is purely text.